Add ability to remove detached internet gateway

branic opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Introduces the ability to remove an internet gateway that is not attached to a VPC. To remove an internet gateway either the ID of the internet gateway or the ID of the attached VPC can be supplied. It is also possible to supply both IDs. If both IDs are supplied then a failure will be generated if the attached VPC ID does not match the user supplied VPC.

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

Fixes #1669 

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
ec2_vpc_igw

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
I wasn't able to add a test for removing an internet gateway that is detached as the module currently doesn't have a way to create a detached internet gateway. I plan on opening a follow-up PR for that use case and will add an appropriate removal test in that PR.

tremble commented on changelogs/fragments/ec2_vpc_igw-delete_unattached-gateway.yml:
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+  - Add ability to delete a vpc internet gateway using the id of the gateway
```
```suggestion
  - ec2_vpc_igw - Add ability to delete a vpc internet gateway using the id of the gateway (https://github.com/ansible-collections/amazon.aws/pull/1786).
```

tremble reviewed (changes requested):
Thanks for taking the time to submit this PR.

I've had a quick look.  The way the documentation's now worded it looks like the module should cope with attaching an existing (unattached) IGW to a VPC, which isn't the case.  I think it's worth either adding this feature or rewording the documentation.  (This might also help with #1787 since you'd at least be able to use the created IGW)

branic commented:
@tremble Please review this PR again when you have a chance. I've added the ability to create an IGW without attaching a VPC to it. I also fixed #1787 and added functionality to attach/detach and change VPC attached to the internet gateway.

branic commented:
@tremble I just wanted to check in and see if you could review this PR again.

Thank you.

tremble commented on plugins/modules/ec2_vpc_igw.py:
```diff
@@ -13,17 +13,36 @@
 author: Robert Estelle (@erydo)
 options:
+  internet_gateway_id:
+    description:
```
```suggestion
    version_added: 7.0.0
    description:
```

tremble commented on plugins/modules/ec2_vpc_igw.py:
```diff
@@ -13,17 +13,36 @@
     default: present
     choices: [ 'present', 'absent' ]
     type: str
+  force_attach:
```
```suggestion
  force_attach:
    version_added: 7.0.0
```

tremble commented on plugins/modules/ec2_vpc_igw.py:
```diff
@@ -234,16 +381,22 @@ def ensure_igw_present(self, vpc_id, tags, purge_tags):
+        ("detach_vpc", True, ("internet_gateway_id", "vpc_id"), True),
+    ]
+
+    module = AnsibleAWSModule(argument_spec=argument_spec, supports_check_mode=True, required_if=required_if)
```
We should throw an error if force_attach and detach_vpc are both set

tremble commented on plugins/modules/ec2_vpc_igw.py:
```diff
@@ -13,17 +13,36 @@
+      - I(vpc_id) must be specified when I(force_attach) is true
+    default: false
+    type: bool
+  detach_vpc:
```
```suggestion
  detach_vpc:
    version_added: 7.0.0
```

tremble reviewed (commented):
Apologies for the delay, life got in the way.

Two issues:
- add version_added to docs for new paramters
- Setting force_attach and detach_vpc to True should throw an error (as early as possible) their behaviours conflict.

branic commented:
Thanks @tremble. I totally understand life getting in the way and I appreciate you taking a look at this so quickly today. I've made the requested changes so hopefully all looks good now. If you find anything else that needs to be addressed I'm happy to do so.

tremble commented:
regate
